controllers/heat_controller.go

```diff
@@ -402,6 +407,20 @@ func (r *HeatReconciler) reconcileNormal(ctx context.Context, instance *heatv1be
 	}
 	secretVars[ospSecret.Name] = env.SetValue(hash)
 
+	// check for custom config secrets
+	for _, secretName := range instance.Spec.CustomServiceConfigSecrets {
```

Comment:
I'm wondering if we really need this check here: in theory we're not setting an hash based on `customServiceConfigSecret`, and if they do not exist, we return an err on L979 (from `generateConfigSecret`) and exit the reconciliation loop because "Config is not ready". That seems a more appropriate error for this missing input. 
However I don't think this is a blocker here, just a general comment on the fact we iterate over the same structure twice in the same reconciliation loop, getting (eventually) the same data as output.
@fao89 thoughts on this?

Reply:
you are right, I'll drop it